src/tools/ts-generator/main.cpp

```diff
@@ -119,9 +159,16 @@ int main(int argc, char* argv[])
         return 1;
     }
 
+    if (settings.allLinks && !settings.linksListToGen.empty())
+    {
+        logs.error() << "Conflicting options, either choose all links or a list";
+        return 1;
+    }
+
     auto study = std::make_shared<Data::Study>(true);
     Data::StudyLoadOptions studyOptions;
     studyOptions.prepareOutput = true;
+    studyOptions.linksLoadTSGen = true;
```

Comment:
About` studyOptions.linksLoadTSGen = true;` : 
We don't need this.
Indeed, imagine we can get a pointer to each link we want to generate TS for, and collect these pointer to a list.
Then we could call **link->loadTSGenTimeSeries(folder)** in the current main program.
The consequence would be : 
- We don't need **studyOptions.linksLoadTSGen** here or elsewhere in the code
- We don't need to call it when running a simulation, simulation that don't need to generate TS for links anyway.
- What else ?

Besides, getting pointer is easy, as we already process study links in function **getLinksToGen(...)**.